src/backup.rs

```diff
@@ -0,0 +1,90 @@
+use crate::error::ClientError;
+use crate::fs::backup::{Backup, ZipBackup};
+use crate::Result;
+
+use chrono::{Local, NaiveDateTime};
+use std::convert::TryFrom;
+use std::path::{Path, PathBuf};
+
+/// Creates a .zip archive from the list of source folders and
+/// saves it to the dest folder.
+pub async fn backup_folders(
+    src_folders: Vec<BackupFolder>,
+    mut dest: PathBuf,
+) -> Result<NaiveDateTime> {
+    let now = Local::now();
+
+    dest.push(format!(
+        "ajour_backup_{}.zip",
+        now.format("%Y-%m-%d_%H-%M-%S")
+    ));
+
+    let zip_backup = ZipBackup::new(src_folders, &dest);
+
+    zip_backup.backup()?;
+
+    let as_of = Archive::try_from(dest)?.as_of;
+
+    Ok(as_of)
+}
+
+/// Finds the latest archive in the supplied backup folder and returns
+/// the datetime it was saved
+pub async fn lastest_backup(backup_dir: PathBuf) -> Option<NaiveDateTime> {
+    let pattern = format!("{}/ajour_backup_[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]_[0-9][0-9]-[0-9][0-9]-[0-9][0-9].zip", &backup_dir.display());
```

Comment:
I like that we find the latest saved rather than saving it to the config. Good thinking here @tarkah.